src/wp-includes/block-patterns.php

```diff
@@ -350,11 +350,24 @@ function _register_theme_block_patterns() {
 				continue;
 			}
 
+			$file_path = $dirpath . $file;
 			// The actual pattern content is the output of the file.
 			ob_start();
-			include $dirpath . $file;
+			@include $file_path;
```

Comment:
I'll leave the decision on the error control operator to others, though it is worth reminding [it is highly discouraged by the WP coding standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#error-control-operator).

I'm not sure why but I see the PHP Coding Standards checks on the related GitHub workflow run with the flag `-n`, which doesn't print warnings. It seems it was the same case also on Travis CI. I'm not sure I understnd why we have coding standards error and warnings when then the warnings are not reported.
@jrfnl and @desrosj may know more.

Instead, commands like `grunt lint:php` run without the `-n` flag and this warning will be a new warning.
I do see in core 22 occurrences of:
`phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged`
but I'm unsure what should be done in this case. However, at the very least there shoud be a comment to explain why the `@` is being used.

Reply:
This is the only way to handle php warning here. Adding this error is silenced, it will result in an empty string being return and caches invalidated.

Reply:
> I'm not sure why but I see the PHP Coding Standards checks on the related GitHub workflow run with the flag -n, which doesn't print warnings. It seems it was the same case also on Travis CI. I'm not sure I understand why we have coding standards error and warnings when then the warnings are not reported.

When automated checking for coding standards issues in PHP files was introduced in WP, everything that could be was bulk fixed. However, there were a significant number of issues (IIRC some 5000 of them) which needed manual review/attention.

To still allow the automated checking to be enabled, if was decided to ignore those for CI by changing them to warnings ( if these weren't warnings already) and ignoring warnings.

Over time, work has been done on these issues, mostly by @SergeyBiryukov, and the number of issues is much much lower now (~600). The intention is to turn "failing on warnings"  on in CI, once the warnings have been brought down to 0.

Reply:
> This is the only way to handle php warning here. Adding this error is silenced, it will result in an empty string being return and caches invalidated.

I don't understand what you are saying here. The `@` should not be used if proper preliminary checks can prevent the warning/error. It should only be used if no amount of checking can prevent a PHP warning/error.

Why would a check to see if the `$file_path` is a file and is readable before including the file not be possible here ?

If such a check was put in place, the error silencing should be able to be removed.

Reply:
@jrfnl I believe what @spacedmonkey was referring to is that in the past there have been several instances where `file_exists()` checks were leading to performance problems. So it's not that it wouldn't be possible, but it's a performance concern. Technically speaking, why check that the file exists _and_ then include the contents when we can also directly include the contents and by that alone recognize whether the file exists? However, I do understand suppressing PHP errors isn't a good approach either.

I have taken a closer look at this locally, and I think adding a `file_exists()` check here is reasonable:
* Past cases of `file_exists()` performance bottlenecks were about checks where the file _didn't_ exist (due to some PHP internals?).
* In the function here, it is expected that those file should exist, the scenario we are addressing here is an edge case. So 99+% of `file_exists()` calls here would be on a file that exists.
* I benchmarked server response time locally with and without a `file_exists()` check and there was no performance difference at all.

So I agree with the approach of adding `file_exists()` as it is a more appropriate solution than suppressing PHP errors, and it appears to _not_ have a relevant performance cost here.

Reply:
Updated in https://github.com/WordPress/wordpress-develop/pull/5462/commits/c244c4e0ca9406ac065118d9ef913ea1b7ca36fb

Reply:
@felixarntz I understand the concern, but performance should never be a reason to introduce sloppy code. I'm happy to see the `file_exists()` added now.

On the note of performance: any time PHP throws an (avoidable) warning/error is actually a performance issue in itself, especially when error logging is on, as the handling of the warning/error via potentially set custom error handlers + the file write for the logging will impact performance.

So even with performance in mind, the general rule of thumb "warnings/errors should be _prevented_ whenever possible instead of silencing them" is still the correct way forward.

Reply:
> I understand the concern, but performance should never be a reason to introduce sloppy code.

Thanks for having a look and for your feedback @jrfnl I'd totally second that.